src/Columns/IColumnImpl.h

```diff
@@ -100,15 +101,8 @@ void IColumn::compareImpl(const Derived & rhs, size_t rhs_row_num,
         if constexpr (use_indexes)
             row = indexes[i];
 
-        int res = compareAt(row, rhs_row_num, rhs, nan_direction_hint);
-
-        /// We need to convert int to Int8. Sometimes comparison return values which do not fit in one byte.
-        if (res < 0)
-            compare_results[row] = -1;
-        else if (res > 0)
-            compare_results[row] = 1;
-        else
-            compare_results[row] = 0;
+        int res = static_cast<const Derived *>(this)->compareAt(row, rhs_row_num, rhs, nan_direction_hint);
```

Comment:
Ok, this is the place of improvement.
But if Column is `final` this call should be already devirtualized?

Reply:
It must be devirtualized.